Update Kubeapps governance and proposal request process

ppbaena opened the pull request:
### Description of the change

Update Kubeapps governance process to include the proposal process to define how to build consensus on a problem statement and solution design before starting work on the implementation.

### Benefits

Unified process and defined lifecycle for new proposals that could impact on Kubeapps. Make the decision process open and transparent for community users and track record of the engineering decisions within the same github repo, so all information is self-contained.

### Possible drawbacks

- Process needs to be followed by maintainers team.
- New template for proposals increase diversity of issues to be opened, so could cause some confusion to users.

### Applicable issues

- related to #4734 

### Additional information
A new board for proposals will be created to track all issues marked with label `kind/proposal`.

antgamdia commented on GOVERNANCE.md:
```diff
@@ -6,50 +6,27 @@ This document defines the project governance for Kubeapps.
 **Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community involved to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
 
-## Code Repositories
+## Code of Conduct
```
Some minor typos:

Not here, but above:

`involved TO simplifying`  -> `involved in simplifying`

And at the "credits" section:

`Sections of this documentS have .... ` --> `Sections of this document have .... `

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+1. Author creates a ["Proposal request"](https://github.com/vmware-tanzu/kubeapps/issues/new?assignees=&labels=kind%2Fproposal&template=proposal-request.md&title=).
+2. Author adds the proposal by creating a PR in draft mode (authors can save their work until ready):
+   - this PR must include a markdown file in the root of the [proposals](./) folder, and
+   - this PR must reference the "Proposal request" (by adding the github ID).
```
```suggestion
   - this PR must reference the "Proposal request" (by adding the GitHub ID).
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+2. Author adds the proposal by creating a PR in draft mode (authors can save their work until ready):
+   - this PR must include a markdown file in the root of the [proposals](./) folder, and
+   - this PR must reference the "Proposal request" (by adding the github ID).
+3. When the author elaborates the proposal sufficiently and consider ready to be reviewed:
```
```suggestion
3. When the author elaborates the proposal sufficiently and considers ready to be reviewed:
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+4. The community reviews the proposal by adding PR reviews in order to mature/converge on the proposal.
+5. When the maintainers reach consensus or supermajority to accept a proposal, they:
+   - change the status of the "Proposal request" to `accepted`,
+   - add the Engineering Decission Record (EDR) to the proposal (in the markdown file) including the following topics:
```
```suggestion
   - add the Engineering Decision Record (EDR) to the proposal (in the markdown file) including the following topics:
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+     4. _decision outcome._
+   - merge the PR, thus adding the new proposal to the `main` branch,
+   - code implementation PRs are submitted separately to implement the solution.
+6. During implementation of an accepted proposal:
```
```suggestion
6. During the implementation of an accepted proposal:
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+   - code implementation PRs are submitted separately to implement the solution.
+6. During implementation of an accepted proposal:
+   - as each implementation PR is created, the "Proposal request" should be updated to link to the new implementation PR, and
+   - when all implementation PRs are merged, the "Proposal Request" should be updated to have status `implemented`, to list all the related PRs, and then, the "Proposal request" should be closed.
```
```suggestion
   - when all the implementation PRs are merged, the "Proposal Request" should be updated to the`implemented` status , to list all the related PRs, and then, the "Proposal request" should be closed.
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+   - when all implementation PRs are merged, the "Proposal Request" should be updated to have status `implemented`, to list all the related PRs, and then, the "Proposal request" should be closed.
+   - If it is discovered that significant unanticipated changes are needed to the proposal, then the implementation work should be paused and the proposal should be updated with the new details to be reviewed by the maintainers again before resuming implementation.
+7. When the maintainers do not reach consensus or supermajority, then the proposal is rejected, and they:
+   - may mark the status of the "Proposal request" to `rejected`,
```
```suggestion
   - may mark the status of the "Proposal request" as `rejected`,
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+
+## Proposal Review
+
+Once a proposal PR marked as "Ready for Review", the community and all Kubeapps maintainers will review the proposal. The goal of the review is to gain an understanding of the problem being solved and the design of the proposed solution.
```
```suggestion
Once a proposal PR is marked as "Ready for Review", the community and all Kubeapps maintainers shall review the proposal. The goal of the review is to gain an understanding of the problem being solved and the design of the proposed solution.
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+
+Maintainers will consider all aspects of the proposed problem and solution, including but not limited to:
+
+- Is the problem within scope for the project?
```
```suggestion
- Is the problem within the scope of the project?
```

antgamdia commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+
+## Maintenance of Accepted Proposal Documents
+
+Proposal documents reflect a point-in-time design and decision. Once approved, they become historical documents, not living documents. There is no expectation that they will be maintained in the future. Instead, significant changes to a feature which came from a previous proposal should be proposed as a fresh proposal. New proposals should link to
```
```suggestion
Proposal documents reflect a point-in-time design and decision. Once approved, they become historical documents, not living documents. There is no expectation that they will be maintained in the future. Instead, significant changes to a feature that came from a previous proposal should be proposed as a fresh proposal. New proposals should link to
```

antgamdia reviewed (commented):
Just minor copy issues. 
Also I noticed this PR is breaking the website preview? Can you run hugo locally to double-check it is just a preview error instead of changes breaking the website?

castelblanque commented on GOVERNANCE.md:
```diff
@@ -6,50 +6,27 @@ This document defines the project governance for Kubeapps.
 
 **Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community involved to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```
```suggestion
**Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community devoted to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```

castelblanque commented on GOVERNANCE.md:
```diff
@@ -6,50 +6,27 @@ This document defines the project governance for Kubeapps.
 
 **Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community involved to simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```
```suggestion
**Kubeapps**, an open-source project, is committed to building an open, inclusive, productive and self-governing open source community focused on simplifying how applications are deployed and managed in Kubernetes clusters. The community is governed by this document to define how the community should work together to achieve this goal.
```

absoludity replied:
+1 for "focused on" rather than the other options.

absoludity replied:
Or "involved in" (only saw that afterwards), both are fine IMO.

absoludity reviewed (approved):
+1 with the feedback from different people here.

ppbaena commented on site/content/docs/latest/reference/proposals/proposals.md:
```diff
@@ -0,0 +1,90 @@
+   - merge the PR, thus adding the new proposal to the `main` branch,
+   - code implementation PRs are submitted separately to implement the solution.
+6. During implementation of an accepted proposal:
+   - as each implementation PR is created, the "Proposal request" should be updated to link to the new implementation PR, and
```
+1 . 
I've added:
`The proposal file (.md file) should be updated by including the version where the proposal was released.`

ppbaena commented:
> Also I noticed this PR is breaking the website preview? Can you run hugo locally to double-check it is just a preview error instead of changes breaking the website?

Done! Thank you Antonio I have updated a folder from design-proposals to proposals and it was not updated in the _index.md file for website. I've updated it and PR should pass all tests now.
